library/Hooks-windows.cpp

```diff
@@ -640,110 +638,96 @@ MORE CRAP
 static void * (*_SDL_CreateSemaphore)(uint32_t initial_value) = 0;
 DFhackCExport void *SDL_CreateSemaphore(uint32_t initial_value)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_CreateSemaphore(initial_value);
 }
 
 static vPtr (*_SDL_CreateThread)(int (*fn)(void *), void *data) = 0;
 DFhackCExport  vPtr SDL_CreateThread(int (*fn)(void *), void *data)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_CreateThread(fn,data);
 }
 
 
 static void (*_SDL_Delay)(uint32_t ms) = 0;
 DFhackCExport void SDL_Delay(uint32_t ms)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     _SDL_Delay(ms);
 }
 
 static void (*_SDL_DestroySemaphore)(void *sem) = 0;
 DFhackCExport void SDL_DestroySemaphore(void *sem)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     _SDL_DestroySemaphore(sem);
 }
 
 static uint8_t (*_SDL_GetAppState)(void) = 0;
 DFhackCExport uint8_t SDL_GetAppState(void)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_GetAppState();
 }
 
 static uint8_t (*_SDL_GetMouseState)(int *, int *) = 0;
 DFhackCExport uint8_t SDL_GetMouseState(int *x, int *y)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_GetMouseState(x,y);
 }
 
 static int (*_SDL_InitSubSystem)(uint32_t flags) = 0;
 DFhackCExport int SDL_InitSubSystem(uint32_t flags)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_InitSubSystem(flags);
 }
 
 static int (*_SDL_SemPost)(void *sem) = 0;
 DFhackCExport int SDL_SemPost(void *sem)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_SemPost(sem);
 }
 
 static int (*_SDL_SemTryWait)(void *sem) = 0;
 DFhackCExport int SDL_SemTryWait(void *sem)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_SemTryWait(sem);
 }
 
 static int (*_SDL_SemWait)(void *sem) = 0;
 DFhackCExport int SDL_SemWait(void *sem)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_SemWait(sem);
 }
 
 static uint32_t (*_SDL_ThreadID)(void) = 0;
 DFhackCExport uint32_t SDL_ThreadID(void)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_ThreadID();
 }
 
 static char* (*_SDL_getenv)(const char *name) = 0;
 DFhackCExport char* SDL_getenv(const char *name)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_getenv(name);
 }
 
 static size_t (*_SDL_strlcat)(char *dst, const char *src, size_t maxlen) = 0;
 DFhackCExport size_t SDL_strlcat(char *dst, const char *src, size_t maxlen)
 {
-    if(!inited)
-        FirstCall();
+    InitSDLPointers();
     return _SDL_strlcat(dst, src, maxlen);
 }
 
-// FIXME: this has to be thread-safe.
-bool FirstCall()
+void FirstCall()
```

Comment:
From https://buildmaster.local.lubar.me/executions/execution-details?executionId=70119#S104.3170

```
Z:/home/buildmaster/work/_E70119/_D0/src/library/Hooks-windows.cpp(740): error C2562: 'FirstCall': 'void' function returning a value
make[2]: *** [library/CMakeFiles/dfhack.dir/Hooks-windows.cpp.obj] Error 2
make[2]: *** Waiting for unfinished jobs....
Z:/home/buildmaster/work/_E70119/_D0/src/library/Hooks-windows.cpp(730): note: see declaration of 'FirstCall'
library/CMakeFiles/dfhack.dir/build.make:998: recipe for target 'library/CMakeFiles/dfhack.dir/Hooks-windows.cpp.obj' failed
```
I can't link to the problem, but it's "return 0" under the "if" block below. In theory, I'd like to keep the return values intact, but we're obviously not using them, and have never used them as far as I know (and nobody has reported the error below), so I guess replacing it with just "return" would work.

Reply:
Done